src/runtime-rs/crates/kata/src/run.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::os::unix::fs::OpenOptionsExt;
+use std::os::unix::io::{FromRawFd, RawFd};
+use std::path::Path;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+
+use containerd_shim_protos::shim::shim_ttrpc;
+use ttrpc::server::Server;
+
+use agent_client::Agent;
+use virtcontainers::{Sandbox, TomlConfig};
+
+use crate::config::load_configuration;
+use crate::event_publisher::EventPublisher;
+use crate::task_service::TaskService;
+use crate::{Error, Result, ShimExecutor, KATA_BIND_FD};
+
+impl ShimExecutor {
+    // Implement ttrpc call from containerd
+    // This functions should be the most outside one for "normal run" command action
+    // So it does not return errors but directly write errors to stderr
+    pub fn run(&mut self) {
+        if let Err(e) = self.args.validate(false) {
+            eprintln!("run shim err: {}", e);
+            return;
+        }
+
+        if let Err(e) = self.long_run() {
+            eprintln!("run shim error: {}", e);
```

Comment:
Inconsistent error message format (`err` / `error`). I'd also consider prefixing with `"ERROR: "` to make it "stand out" (and more consistent for `grep`-ability ;)

It _might_ be worth adding a `self.real_run()` or similar so that this method could become something like:

```rust
pub fn run(&mut self) {
    if let Err(e) = self.real_run() {
        eprintln!("ERROR: {:#}", e);
    }
}
```

That looks simpler and also means that `real_run()` can be unit tested more easily.